packages/gatsby-link/src/index.js

```diff
@@ -89,26 +75,33 @@ class GatsbyLink extends React.Component {
     if (this.state.IOSupported && ref) {
       // If IO supported and element reference found, setup Observer functionality
       handleIntersection(ref, () => {
-        ___loader.enqueue(this.state.to.pathname)
+        ___loader.enqueue(parsePath(this.props.to).pathname)
       })
     }
   }
 
   render() {
-    const { onClick, onMouseEnter, ...rest } = this.props
-    let El
-    if (Object.keys(NavLinkPropTypes).some(propName => this.props[propName])) {
-      El = NavLink
+    const { onClick, onMouseEnter, location, ...rest } = this.props
+    let getProps
+    if (this.props.getProps) {
+      getProps = this.props.getProps
     } else {
-      El = Link
+      getProps = ({ isCurrent }) =>
+        isCurrent
+          ? {
+              className: this.props.activeClassName,
```

Comment:
I think we need to add default `className` (similiar how styles are merged in line below)
`react-router`'s `NavLink` does that - https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/modules/NavLink.js#L41-L42 if we want to maintain backward compatibility

Reply:
Ah good catch, fixed in https://github.com/gatsbyjs/gatsby/pull/6918/commits/33583a24548af0f47ad54f00497f3675dd3c9963